File: python/hamming/hamming.py
"""Compute the hamming distance."""

def distance(strand_a: str, strand_b: str) -> int:
    if len(strand_a) != len(strand_b):
        raise ValueError("Strands must be of equal length.")
    return sum(a != b for a, b in zip(strand_a, strand_b))

File: javascript/hamming/hamming.js
export const compute = (strandA, strandB) => {
  if (strandA.length !== strandB.length) {
    throw new Error('strands must be of equal length');
  }
  let diff = 0;
  for (let i = 0; i < strandA.length; i++) {
    if (strandA[i] !== strandB[i]) {
      diff++;
    }
  }
  return diff;
};
